.github/actions/setup-go/action.yaml

```diff
@@ -1,36 +1,62 @@
 name: "Setup Go"
 description: |
   Sets up the Go environment for tests, builds, etc.
+inputs:
+  version:
+    description: "The Go version to use."
+    default: "1.20.5"
 runs:
   using: "composite"
   steps:
+    - name: Cache go toolchain
+      uses: buildjet/cache@v3
+      with:
+        path: |
+          ${{ runner.tool_cache }}/go/${{ inputs.version }}
+        key: gotoolchain-${{ runner.os }}-${{ inputs.version }}
+        restore-keys: |
+          gotoolchain-${{ runner.os }}-
+
     - uses: buildjet/setup-go@v4
       with:
-        cache: true
-        go-version: "1.20.5"
+        # We do our own caching for implementation clarity.
+        cache: false
+        go-version: ${{ inputs.version }}
+
+    - name: Get cache dirs
+      shell: bash
+      run: |
+        set -x
+        echo "GOMODCACHE=$(go env GOMODCACHE)" >> $GITHUB_ENV
+        echo "GOCACHE=$(go env GOCACHE)" >> $GITHUB_ENV
+
+    # We split up GOMODCACHE from GOCACHE because the latter must be invalidated
+    # on code change, but the former can be kept.
+    - name: Cache $GOMODCACHE
+      uses: buildjet/cache@v3
+      with:
+        path: |
+          ${{ env.GOMODCACHE }}
+        key: gomodcache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}-${{ github.job }}
+        restore-keys: |
+          gomodcache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}-
+          gomodcache-${{ runner.os }}-
 
-    - name: Cache go
+    - name: Cache $GOCACHE
       uses: buildjet/cache@v3
       with:
-        # ~/go/pkg is the same across operating systems.
         path: |
-          ~/go/pkg
-          ~/.cache/go-build
-          ~/AppData/Local/go-build
-          ~/Library/Caches/go-build
+          ${{ env.GOCACHE }}
         # Job name must be included in the key for effective
         # test cache reuse.
-        key: go-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/*.go', 'go.**') }}
+        key: gocache-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/*.go', 'go.**') }}
         restore-keys: |
-          go-${{ runner.os }}-${{ github.job }}-
-          go-${{ runner.os }}-
-          go-
+          gocache-${{ runner.os }}-${{ github.job }}-
```

Comment:
Hyphen at the end?

Reply:
That was on purpose. It's how I've seen it done on a few docs. I guess it clearly symbolizes the end of a token.